### PR TITLE
Rename sectionName to sectionId

### PR DESCRIPTION
**Pull request by tomrf1**
In DCR we get two section fields from Frontend for articles:
- `article.config.section`
- `article.sectionName`

Despite the difference in name, they are always the same, and they are in fact the section _ID_. E.g. `australia-news`.
This contradicts the CAPI content model, e.g:
Screenshot 2023-07-11 at 09 33 13

Where we are currently using `sectionName`, we're actually using it as an ID.

Currently both fields are used in different places. This PR: 
- changes usages of `article.sectionName` to `article.config.section`, for consistency
- renames any props/variables called `section` or `sectionName` to `sectionId`, for clarity

### CODE REVIEW

File: dotcom-rendering/src/components/StickyBottomBanner.importable.tsx [hunk 1]
```diff
@@ -292,7 +289,7 @@ export const StickyBottomBanner = ({
 			idApiUrl,
 		});
 		const brazeArticleContext: BrazeArticleContext = {
-			section: sectionName,
+			section: sectionId,
```
Comment (tomrf1) [inline dotcom-rendering/src/components/StickyBottomBanner.importable.tsx:292]: `BrazeArticleContext` is defined in an npm library, so I'm not changing that field name now

File: dotcom-rendering/src/components/FrontMostViewed.tsx [hunk 2]
```diff
@@ -91,7 +91,7 @@ export const FrontMostViewed = ({
 			) : (
 				<MostViewedFooter
 					tabs={tabs}
-					sectionName="Most viewed"
+					sectionId="Most viewed"
```
Comment (tomrf1) [inline dotcom-rendering/src/components/FrontMostViewed.tsx:94]: This one is a bit odd. For articles we really do pass the sectionId into `MostViewedFooter`, but for fronts we pass in "Most viewed".
It's actually only used to build a `data-link-context` attribute in a child component.

File: dotcom-rendering/src/components/ArticleBody.tsx [hunk 3]
```diff
@@ -26,13 +26,12 @@ type Props = {
 	ajaxUrl: string;
 	isAdFreeUser: boolean;
 	switches: Switches;
-	section: string;
+	sectionId: string;
```
Comment (tjmw) [inline dotcom-rendering/src/components/ArticleBody.tsx:29]: Really good decision to rename this 👍

**Review by tjmw — APPROVED** [review]
Excellent!

**Pull request merged**